test/rpc_test.py

```diff
@@ -1091,6 +1091,51 @@ def test_call_method_on_rref(self):
 
         self.assertEqual(result, sum(vals))
 
+    def _test_rref_leak(self, ignore_leak=False):
+        rpc.init_rpc(
+            name="worker{}".format(self.rank),
+            backend=self.rpc_backend,
+            init_method=self.init_method,
+            rank=self.rank,
+            world_size=self.world_size,
+            rpc_agent_options=self.rpc_agent_options,
+        )
+
+        # This is for the below `dist.barrier`.
+        # For `RpcAgent` other than `ProcessGroupAgent`,
+        # no `_default_pg` is initialized.
+        if not dist.is_initialized():
+            dist.init_process_group(
+                backend="gloo",
+                init_method=self.init_method,
+                rank=self.rank,
+                world_size=self.world_size,
+            )
+        # Wait for all init to complete.
+        dist.barrier()
+
+        rref = rpc.remote(
+            "worker{}".format((self.rank + 1) % self.world_size),
+            torch.add,
+            args=(torch.ones(2, 2), 1)
+        )
+
+        if ignore_leak:
+            import torch.distributed.rpc.api as api
+            api._ignore_rref_leak = True
+            rpc.wait_all_workers()
+        else:
+            with self.assertRaisesRegex(RuntimeError, "Leaking RRef"):
```

Comment:
Nit, I think this assert should be part of `test_ignore_rref_leak`

Reply:
Could you please elaborate? Why `test_ignore_rref_leak` would expect a runtime error? Or do you mean sth like `self.assertWarnsRegex` after we add the warning?

Reply:
My bad, I mean move to `test_rref_leak`